### PR TITLE
Incorrect number of failed tests (Issue #132) 

### PR DESCRIPTION
**Pull request by Beidendorfer**

### CODE REVIEW

**Comment by sagatowski** [comment]
Do you have a sample program that I can use to verify that this is working, preferably with many tests?

**Review by sagatowski — COMMENTED** [review]
Thanks for your commits. Here are some review comments.

File: TcUnit/TcUnit/POUs/FB_TestSuite.TcPOU [hunk 1]
```diff
@@ -62,6 +62,7 @@ VAR
     LowerCasedTestName : Tc2_System.T_MaxString;
     TrimmedTestName : Tc2_System.T_MaxString;
     IgnoreCurrentTestCase : BOOL;
+	NumberOfTestsToAnalyse: UINT;
```
[the remark this answers is not in the document]
Comment (Beidendorfer) [reply TcUnit/TcUnit/POUs/FB_TestSuite.TcPOU]: done in c4db0ad

File: TcUnit/TcUnit/POUs/FB_TestResults.TcPOU [hunk 2]
```diff
@@ -12,6 +12,7 @@ VAR
     StoringTestSuiteTrigger : Tc2_Standard.R_TRIG;
     StoredTestSuiteResults : BOOL;
     StoredGeneralTestResults : BOOL;
+	NumberOfTestsToAnalyse: UINT;
```
[the remark this answers is not in the document]
Comment (Beidendorfer) [reply TcUnit/TcUnit/POUs/FB_TestResults.TcPOU]: done in c4db0ad

File: TcUnit/TcUnit/POUs/FB_AdsTestResultLogger.TcPOU [hunk 3]
```diff
@@ -37,9 +37,11 @@ END_VAR
     <Method Name="LogTestSuiteResults" Id="{3263938a-5e4f-4e5f-bfa0-9c9469439d97}">
       <Declaration><![CDATA[METHOD PUBLIC LogTestSuiteResults
 VAR
-    TcUnitTestResults : REFERENCE TO ST_TestSuiteResults;
-    StringToPrint : Tc2_System.T_MaxString; 
-    TestsInTestSuiteCounter : UINT(1..GVL_Param_TcUnit.MaxNumberOfTestsForEachTestSuite);
+    TcUnitTestResults             : REFERENCE TO ST_TestSuiteResults;
+    StringToPrint                 : Tc2_System.T_MaxString; 
+    TestsInTestSuiteCounter       : UINT(1..GVL_Param_TcUnit.MaxNumberOfTestsForEachTestSuite);
+    NumberMaxTestToPrint          : UINT(1..GVL_Param_TcUnit.MaxNumberOfTestsForEachTestSuite);
```
[the remark this answers is not in the document]
Comment (Beidendorfer) [reply TcUnit/TcUnit/POUs/FB_AdsTestResultLogger.TcPOU]: done in c4db0ad

File: TcUnit/TcUnit/POUs/FB_AdsTestResultLogger.TcPOU [hunk 4]
```diff
@@ -37,9 +37,11 @@ END_VAR
     <Method Name="LogTestSuiteResults" Id="{3263938a-5e4f-4e5f-bfa0-9c9469439d97}">
       <Declaration><![CDATA[METHOD PUBLIC LogTestSuiteResults
 VAR
-    TcUnitTestResults : REFERENCE TO ST_TestSuiteResults;
-    StringToPrint : Tc2_System.T_MaxString; 
-    TestsInTestSuiteCounter : UINT(1..GVL_Param_TcUnit.MaxNumberOfTestsForEachTestSuite);
+    TcUnitTestResults             : REFERENCE TO ST_TestSuiteResults;
```
[the remark this answers is not in the document]
Comment (Beidendorfer) [reply TcUnit/TcUnit/POUs/FB_AdsTestResultLogger.TcPOU]: done in c4db0ad

File: TcUnit/TcUnit/POUs/FB_AdsTestResultLogger.TcPOU [hunk 5]
```diff
@@ -79,8 +81,18 @@ IF PrintingTestSuiteResultNumber <= GVL_TcUnit.NumberOfInitializedTestSuites AND
                                             MsgFmtStr := StringToPrint,
                                             StrArg := '');
 
+        (* Print Error if there are too many tests in the Test Suite*)
+        IF (TcUnitTestResults.TestSuiteResults[PrintingTestSuiteResultNumber].NumberOfTests > GVL_Param_TcUnit.MaxNumberOfTestsForEachTestSuite) THEN
+            NumberMaxTestToPrint := GVL_Param_TcUnit.MaxNumberOfTestsForEachTestSuite;
+            GVL_TcUnit.AdsMessageQueue.WriteLog(MsgCtrlMask := Tc2_System.ADSLOG_MSGTYPE_ERROR,
+                                                MsgFmtStr := concat('| ',concat(TO_STRING(TcUnitTestResults.TestSuiteResults[PrintingTestSuiteResultNumber].NumberOfTests -  GVL_Param_TcUnit.MaxNumberOfTestsForEachTestSuite) ,' Tests failed because = %s')),
+                                                StrArg := concat (TO_STRING(TcUnitTestResults.TestSuiteResults[PrintingTestSuiteResultNumber].NumberOfTests) ,concat(' in this Test-Suite : ',concat(TcUnitTestResults.TestSuiteResults[PrintingTestSuiteResultNumber].Name, '. It is more as in the GVL_Param_TcUnit configured'))));
```
[the remark this answers is not in the document]
Comment (Beidendorfer) [reply TcUnit/TcUnit/POUs/FB_AdsTestResultLogger.TcPOU]: done in 5b99649

> Why start with character "|"?

All errors starts with "|" 
image

**Comment by Beidendorfer** [comment]
> Do you have a sample program that I can use to verify that this is working, preferably with many tests?

My plan was to write tests in Verifier.
In the PLC part it is easy, I just have to look at the analysis in Verifier.DotNet.
How it works there

Should I first write an example program to test? otherwise I would try to update the verifier right away.

**Comment by sagatowski** [comment]
Merged. Thanks for your contribution.

**Pull request merged**